### ISSUE TITLE
Make `contentType` configurable or auto-detected

### ISSUE THREAD

**Issue by elidupuis** [ember-cli-deploy/ember-cli-deploy-s3-index#45]
**TL;DR**
I can upload any file type via the `filePattern` option, but the `contentType` is always `text/html`.

**Rationale**
I know this is the **s3-_index_** plugin—which implies `index.html`—but I'm wondering how you feel about make the `contentType` configurable (and/or automatically detected): https://github.com/ember-cli-deploy/ember-cli-deploy-s3-index/blob/master/lib/s3.js#L50.

The reason I ask I because I'd like to mimic the `index.html:rev123` approach of the Lightning Strategy except with a tarball of the `deploy-dist` directory instead of `index.html`. My use case is an AWS OpsWorks setup (essentially Chef recipes) which grabs the tarball from S3 and then runs various other AWS deployment tasks. Controlling the `contentType` would allow `ember-cli-deploy-revision-data` to work in the same fashion as it does with the Lightning strategy. I chatted briefly about this in the Ember Slack channel last week with @achambers and @lukemelia.

Obviously, because we need the Chef recipes to run, our strategy lacks the instantaneous deployment of the Lightning Strategy but it still shares some of the benefits. I've started thinking of it as the Thunder Strategy. :stuck_out_tongue_winking_eye: 

Thoughts? Concerns?

**Comment by LevelbossMike** [comment]
Closing after merge of #46
